packages/engine/src/worker/runner/javascript/mod.rs

```diff
@@ -311,75 +446,144 @@ fn batches_from_shared_store(
     })
 }
 
-fn mem_batch_to_js<'m>(
-    mv8: &'m MiniV8,
+fn mem_batch_to_js<'s>(
+    scope: &mut v8::HandleScope<'s>,
     batch_id: &str,
-    mem: mv8::Object<'m>,
+    mem: v8::Local<'s, v8::Object>,
     persisted: Metaversion,
-) -> Result<mv8::Value<'m>> {
-    let batch = mv8.create_object();
-    let batch_id = mv8.create_string(batch_id);
-    batch.set("id", mv8::Value::String(batch_id))?;
-    batch.set("mem", mem)?;
-    batch.set("mem_version", persisted.memory())?;
-    batch.set("batch_version", persisted.batch())?;
-    Ok(mv8::Value::Object(batch))
+) -> Result<v8::Local<'s, v8::Value>> {
+    let batch = v8::Object::new(scope);
+    let batch_id = new_js_string(scope, batch_id)?;
+
+    let id_field = new_js_string(scope, "id")?;
+    let mem_field = new_js_string(scope, "mem")?;
+    let mem_version_field = new_js_string(scope, "mem_version")?;
+    let batch_version_field = new_js_string(scope, "batch_version")?;
+
+    batch
+        .set(scope, id_field.into(), batch_id.into())
+        .ok_or_else(|| Error::V8("Could not set id field on batch".to_string()))?;
+    batch
+        .set(scope, mem_field.into(), mem.into())
+        .ok_or_else(|| Error::V8("Could not set mem field on batch".to_string()))?;
+    let js_memory = v8::Number::new(scope, persisted.memory() as f64);
+    batch
+        .set(scope, mem_version_field.into(), js_memory.into())
+        .ok_or_else(|| Error::V8("Could not set mem_version field on batch".to_string()))?;
+    let js_batch = v8::Number::new(scope, persisted.batch() as f64);
+    batch
+        .set(scope, batch_version_field.into(), js_batch.into())
+        .ok_or_else(|| Error::V8("Could not set batch_version field on batch".to_string()))?;
+
+    Ok(batch.into())
 }
 
-fn batch_to_js<'m>(
-    mv8: &'m MiniV8,
+fn batch_to_js<'s>(
+    scope: &mut v8::HandleScope<'s>,
     mem: &Memory,
     persisted: Metaversion,
-) -> Result<mv8::Value<'m>> {
-    // TODO: Is `mem.data.len()` different from `mem.size`? (like Vec capacity vs len?)
-    let arraybuffer = mv8.create_arraybuffer(mem.data.as_ptr(), mem.size);
+) -> Result<v8::Local<'s, v8::Value>> {
+    // The memory is owned by the shared memory, we don't want JS or Rust to try to de-allocate it
+    unsafe extern "C" fn no_op(_: *mut std::ffi::c_void, _: usize, _: *mut std::ffi::c_void) {}
+
+    // https://github.com/denoland/rusty_v8/pull/926
+    //
+    // # Safety
+    //
+    // `mem.data` points to valid memory and is valie for `mem.size` bytes
+    // `no_op` will not try to de-allocate share memory
+    //
+    // As of writing this comment I don't know if/when shared memory is accessed/modified
+    // there could be some unsafety
+    // It's also not 100% clear what [`ArrayBuffer`] expects
+    // (is it ok to read/write while the [`ArrayBuffer`] exists?)
```

Comment:
```suggestion
    // SAFETY: `mem.data` points to valid memory and is valid for `mem.size` bytes `no_op` will not
    //         try to de-allocate share memory.
    // TODO: Investigate, that this does not have any implications on reading/writing shared memory.
    //       It's also not 100% clear what [`ArrayBuffer`] expects, is it ok to read/write while the
    //       [`ArrayBuffer`] exists?)
```

Reply:
Addressed in https://github.com/hashintel/hash/pull/414/commits/70b8bbf11bb01ae88609a4351f0399ac4df01c53

Reply:
Thank you. I noticed that there are links, too. Could you remove those as well?

Reply:
Addressed in https://github.com/hashintel/hash/pull/414/commits/91d5610d0f0f00a733dff734271d9fac9f38e6b7